### PR TITLE
[Fix] fix config and code errors of swin/hrformer

### PR DESCRIPTION
**Pull request by ly015**
## Modification

- Fix misplaced `paramwise_cfg` in configs
- Fix a bug in swin backbone initialization

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by Ben-Louis** [comment]
There are still some discrepancies between the Swin model and the checkpoint from the previous version.

```
unexpected key in source state_dict: backbone.norm0.weight, backbone.norm0.bias, backbone.norm1.weight, backbone.norm1.bias, backbone.norm2.weight, backbone.norm2.bias
```

Is this behaving as expected?

**Comment by Ben-Louis** [comment]
> There are still some discrepancies between the Swin model and the checkpoint from the previous version.
> 
> The following error is reported when initializing the model defined by `td-hm_swin-t-p4-w7_8xb32-210e_coco-256x192`:
> 
> ```
> unexpected key in source state_dict: backbone.norm0.weight, backbone.norm0.bias, backbone.norm1.weight, backbone.norm1.bias, backbone.norm2.weight, backbone.norm2.bias
> ```
> 
> Is this behaving as expected?

I get it. Only the normalization layers that are necessary for the `out_indices` are constructed.

File: mmpose/models/backbones/swin.py [hunk 1]
```diff
@@ -668,13 +667,11 @@ def init_weights(self, pretrained=None):
                 and self.init_cfg['type'] == 'Pretrained'):
             # Suppress zero_init_residual if use pretrained model.
             logger = get_root_logger()
-            _state_dict = get_state_dict(
+            state_dict = get_state_dict(
                 self.init_cfg['checkpoint'], map_location='cpu')
             if self.convert_weights:
                 # supported loading weight from original repo,
```
Comment (Ben-Louis) [inline mmpose/models/backbones/swin.py:673]:
```suggestion
                # supported loading weight from original repo
```

**Review by Ben-Louis — APPROVED** [review]

**Review by Tau-J — APPROVED** [review]
